docs/tutorials/3_The_Part_In_Between.rst

```diff
@@ -0,0 +1,196 @@
+to do this and I'm not gonna tell you which one is best but I'll show
+you a couple.
+
+One way is to explicitly change the array's type after you make it and
+before you write it to the simulation the first time:
+
+.. code-block:: python
+
+    ages = np.random.randint(1, 100, size=population_size).astype(float)
+
+Another is to generate random floats directly, which is closer to what
+you might do if you were trying to generate ages that matched some
+distribution:
+
+.. code-block:: python
+
+    ages = np.random.random(size=population_size) * 99 + 1
+
+Make one of those changes (or even try modeling a more meaningful
+distribution using np.random.normal or similar) and run it again and
+things should go. If you stick the ``inspector`` back in you should
+see people aging as time passes.
+
+Using (Fake) Real Data
+----------------------
+
+Alright, now people are getting older but that doesn't really mean
+much because our mortality rate is static. What we want is for the
+mortality rate to depend on properties of the simulants. For now our
+only properties are age and sex so we'll start with those. If you
+happen to have a csv file of realistic mortality rates by age and sex
+lying around (and if you're the kind of person who's interested in
+Vivarium, you probably do) then use that. If not, you can make one
+up. It's easy:
+
+.. code-block:: python
+
+    import pandas as pd
+    import numpy as np
+
+    rows = []
+    for year in range(1990, 2015):
+            for age in range(150):
+                for sex in ['Female', 'Male']:
+                    rows.append([year, age, sex, np.random.random()])
+    df = pd.DataFrame(rows, columns=['year', 'age', 'sex', 'rate'])
+    df.to_csv('mortality_rate.csv', index=False)
+
+That just loops through all the permutations of age and sex that we
+might need and assigns a random number to each dumps it to disk where
+we can read it later. Obviously, in a real model you'd want to get the
+data from a more reputable source than a random number generator.
+
+With this reference data we can go back to our mortality code and get
+rid of that static rate. The first step is to register the reference
+data with the simulation. This isn't strictly necessary but having the
+simulation manage the data for us lets it do some optimizations that
+make access faster than it would be if we just loaded a pandas
+dataframe and used it directly. First we go to the ``Mortality``
+class's ``setup`` method and tweak it to load in the mortality rate
+table:
+
+.. code-block:: python
+
+    def setup(self, builder):
+        self.mortality_rate = builder.rate('mortality_rate')
+        self.rate_table = builder.lookup(pd.read_csv('mortality_rate.csv'))
+
+The ``builder.lookup`` method takes a ``DataFrame`` and loads it into
+the internal reference data system returning a function that can be
+used to access it by population index later (the same indexes we've
+been using to access and update the population table). The
```

Comment:
I think we should be explicit about the fact that we're creating an interpolation over the data here.